src/solvers/dgsem_structured/dg_2d.jl

```diff
@@ -264,6 +264,108 @@ end
     return nothing
 end
 
+@inline function calcflux_fvO2!(fstar1_L, fstar1_R, fstar2_L, fstar2_R, u,
+                              mesh::StructuredMesh{2},
```

Comment:
This routine should now work for `mesh::Union{StructuredMesh{2}, StructuredMeshView{2},
                                          UnstructuredMesh2D,
                                          P4estMesh{2}, T8codeMesh{2}}`
                                          
see

https://github.com/trixi-framework/Trixi.jl/blob/7b3316f36a6363e8997e1a138c08398a5b6059b2/src/solvers/dgsem_structured/dg_2d.jl#L290-L293

Could you also include those mesh types and maybe check (does not need to be a full convergence study) if the results look reasonable ? Maybe you can look for tests where they use the first order FV scheme and just replace that with the second order scheme with inner reconstruction mode.

Reply:
Documenting these results should be fine, i.e., just provide a comparison of the errors or plots in the PR. 

If you want, though, you could add special tests similar to these:

https://github.com/trixi-framework/Trixi.jl/blob/26d86008edd748425e75c80a2bbb748334b02728/test/test_structured_2d.jl#L264-L265

I.e., take an existing elixir and then exchange the solver to the O2 one. Then we do not need to add a new elixir, but test anyways.